EventFilter/CTPPSRawToDigi/plugins/TotemVFATRawToDigi.cc

```diff
@@ -168,9 +169,19 @@ template <typename DigiType>
 void TotemVFATRawToDigi::run(edm::Event &event, const edm::EventSetup &es) {
   // get DAQ mapping
   ESHandle<TotemDAQMapping> mapping = es.getHandle(totemMappingToken);
+  if (!mapping) {
+    LogError("TotemVFATRawToDigi") << "TotemVFATRawToDigi: no mapping found for " << subSystemName;
+  }
 
   // get analysis mask to mask channels
-  ESHandle<TotemAnalysisMask> analysisMask = es.getHandle(analysisMaskToken);
+  TotemAnalysisMask analysisMask;
+  ESHandle<TotemAnalysisMask> analysisMaskHandle = es.getHandle(analysisMaskToken);
+  if (analysisMaskHandle) {
```

Comment:
as `analysisMaskHandle` is an object on the stack, what's the point of this `if` statement? 
I think what you want is instead to check the validity of the `ESHandle` instead.

Reply:
The idea was to check if the handle containing TotemAnalysisMask was empty (for example when no TotemAnalysisMask record was available). If it was empty we would like to create "empty" analysis mask, otherwise we try to retrieve it from handle.

Reply:
Well, clearly the test results this does not work.

Reply:
We can cross-check again, but I believe the tests are failing due to missing `TotemReadoutRcd`, not due to missing `TotemAnalysisMask`. Right now both are missing in the CondDB (waiting for GT). We can safely handle missing `TotemAnalysisMask` by creating an empty object. 

On the other hand missing `TotemReadoutRcd` is a major issue. In that case the exception is raised, as expected. Without DAQ mapping (stored in `TotemReadoutRcd`) we won't be able to run the RAW2DIGI.

Reply:
> We can cross-check again, but I believe the tests are failing due to missing

I don't think the current pattern will work.

Reply:
Can you suggest us something better ?

Reply:
I already did above: https://github.com/cms-sw/cmssw/pull/42711/files#r1315726789.
E.g. using

https://github.com/cms-sw/cmssw/blob/0295cb18cb10e5774d20c2d910e4dc17ed352357/FWCore/Framework/interface/ESHandle.h#L44

or 
https://github.com/cms-sw/cmssw/blob/0295cb18cb10e5774d20c2d910e4dc17ed352357/FWCore/Framework/interface/ESHandle.h#L46

Reply:
> I already did above: https://github.com/cms-sw/cmssw/pull/42711/files#r1315726789. E.g. using

Thank you, that is clear now, we will try it.

Reply:
I have changed the validity checks